pkg/daemon/ceph/osd/volume.go

```diff
@@ -292,6 +289,25 @@ func (a *OsdAgent) initializeBlockPVC(context *clusterd.Context, devices *Device
 				immediateExecuteArgs = append(immediateExecuteArgs, []string{crushDeviceClassFlag, crushDeviceClass}...)
 			}
 
+			if isEncrypted {
+				immediateExecuteArgs = append(immediateExecuteArgs, encryptedFlag)
+
+				// Generate dmcrypt key
+				key, err := generateDmCryptKey()
```

Comment:
The operator should be generating the encryption key and storing it in the secret. Then the osd prepare and daemon pods will just mount the secret, and no need to add the RBAC for the osds to access all secrets.